x/metadata/types/specification_test.go

```diff
@@ -0,0 +1,497 @@
+				"Unit Tests",
+				"",
+				"http://www.test.com",
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid website url - http at max length",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"http://" + strings.Repeat("f", maxURLLength - 7),
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid website url - https",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"https://www.test.com",
+				"",
+			),
+			"",
+			false,
+		},
+		{
+			"valid website url - https at max length",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"https://" + strings.Repeat("s", maxURLLength - 8),
+				"",
+			),
+			"",
+			false,
+		},
+
+		// Icon url tests
+		{
+			"invalid icon url - too long",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				strings.Repeat("h", maxURLLength + 1),
+			),
+			fmt.Sprintf("url IconUrl exceeds maximum length (expected <= %d got: %d)", maxURLLength, maxURLLength + 1),
+			true,
+		},
+		{
+			"invalid icon url - no protocol",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"www.test.com",
+			),
+			fmt.Sprintf("url IconUrl must begin with either http:// or https://"),
+			true,
+		},
+		{
+			"valid icon url - http",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"http://www.test.com",
+			),
+			"",
+			false,
+		},
+		{
+			"valid icon url - http at max length",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"http://" + strings.Repeat("f", maxURLLength - 7),
+			),
+			"",
+			false,
+		},
+		{
+			"valid icon url - https",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"https://www.test.com",
+			),
+			"",
+			false,
+		},
+		{
+			"valid icon url - https at max length",
+			NewDescription(
+				"Unit Tests",
+				"",
+				"",
+				"https://" + strings.Repeat("s", maxURLLength - 8),
+			),
+			"",
+			false,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		s.T().Run(tt.name, func(t *testing.T) {
+			err := tt.desc.ValidateBasic("")
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Scope ValidateBasic error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantErr {
+				require.Equal(t, tt.want, err.Error())
+			}
+		})
+	}
+}
+
+func (s *specificationTestSuite) TestScopeSpecString() {
+	s.T().Run("scope specification string", func(t *testing.T) {
+		scopeSpecUuid := uuid.MustParse("c2074a03-6f6d-4029-bfe2-c3a5eb7e68b1")
+		contractSpecUuid := uuid.MustParse("540dadf1-3dbc-4c3f-a205-7575b7f74384")
+		scopeSpec := NewScopeSpecification(
+			ScopeSpecMetadataAddress(scopeSpecUuid),
+			NewDescription(
+				"TestScopeSpecString Description",
+				"This is a description of a description used in a unit test.",
+				"https://figure.com/",
+				"https://figure.com/favicon.png",
+			),
+			[]string{specTestBech32},
+			[]PartyType{PartyType_PARTY_TYPE_OWNER},
+			[]MetadataAddress{GroupSpecMetadataAddress(contractSpecUuid)},
+		)
+		expected := "specification_id: scopespec1qnpqwjsrdak5q2dlutp6t6m7dzcscd7ff6\n" +
```

Comment:
Escaped multi-line strings are easier to work with and read in these situations in my opinion.